services/datamanager/data_manager.go

```diff
@@ -61,21 +51,68 @@ func FromDependencies(deps resource.Dependencies, name string) (Service, error)
 	return resource.FromDependencies[Service](deps, Named(name))
 }
 
-// DataCaptureConfigs specify a list of methods to capture on resources.
-type DataCaptureConfigs struct {
-	CaptureMethods []DataCaptureConfig `json:"capture_methods"`
+// AssociatedConfig specify a list of methods to capture on resources and implements the resource.AssociatedConfig interface.
+type AssociatedConfig struct {
+	CaptureMethods []*DataCaptureConfig `json:"capture_methods"`
+}
+
+func newAssociatedConfig(attributes utils.AttributeMap) (*AssociatedConfig, error) {
+	md, err := json.Marshal(attributes)
+	if err != nil {
+		return nil, err
+	}
+	var conf AssociatedConfig
+	if err := json.Unmarshal(md, &conf); err != nil {
+		return nil, err
+	}
+	return &conf, nil
+}
+
+// Equals describes if an DataCaptureConfigs is equal to a given AssociatedConfig.
+func (ac *AssociatedConfig) Equals(other resource.AssociatedConfig) bool {
+	ac2, err := utils.AssertType[*AssociatedConfig](other)
+	if err != nil {
+		return false
+	}
+	if len(ac.CaptureMethods) != len(ac2.CaptureMethods) {
+		return false
+	}
+	for i := 0; i < len(ac.CaptureMethods); i++ {
+		if !ac.CaptureMethods[i].Equals(ac2.CaptureMethods[i]) {
+			return false
+		}
+	}
+	return true
 }
 
 // UpdateResourceNames allows the caller to modify the resource names of data capture in place.
-func (dcs *DataCaptureConfigs) UpdateResourceNames(updater func(old resource.Name) resource.Name) {
-	for idx := range dcs.CaptureMethods {
-		dcs.CaptureMethods[idx].Name = updater(dcs.CaptureMethods[idx].Name)
+func (ac *AssociatedConfig) UpdateResourceNames(updater func(old resource.Name) resource.Name) {
+	for idx := range ac.CaptureMethods {
+		ac.CaptureMethods[idx].Name = updater(ac.CaptureMethods[idx].Name)
+	}
+}
+
+// Link associates an AssociatedConfig to a specific resource model (e.g. builtin data capture).
+func (ac *AssociatedConfig) Link(conf *resource.Config) {
```

Comment:
would it make sense to make this part of the interface?

Reply:
At first I didn't think this would do anything but it actually cleans things up a lot.  Good idea!

Reply:
Added